run_perf_tests.py

```diff
@@ -307,25 +320,43 @@ def analyze_data_files(dest_dir):
         return 1
 
 
-def get_testcase_name(dest_dir, full_filename):
-    dump_results = parse_dump_file(full_filename)
-    testcase_name = dump_results['Testname']
-    testcase_names.append(testcase_name)
-
-
-def xml(dest_dir, jank_perc, execution_time):
-    xml_file_dir = os.path.join(dest_dir, 'app/build/outputs/androidTest-results/connected')
+def xml(dest_dir,device_dir):
+    xml_file_dir = os.path.join(dest_dir, 'app', 'build', 'outputs', 'androidTest-results', 'connected')
     for file in os.listdir(xml_file_dir):
         xml_file_name = file
-    tree = ElementTree.ElementTree(file = xml_file_dir + '/' + xml_file_name)
+    tree = ElementTree.ElementTree(file = os.path.join(xml_file_dir, xml_file_name))
+    tree.write("/tmp/results-debug.xml")
+
     for element in tree.findall('testcase'):
         name = element.get('name')
-        for testcase_name, jank_perc, execution_time in zip(testcase_names,list_for_jank_perc,list_for_execution_time):
-            if name == testcase_name:
-                element.set('jank-percentage', jank_perc)
-                element.set('execution-time', execution_time)
+        classname = element.get('classname')
+        folder_name = classname + '_' + name
+        test_data_dir = os.path.join(device_dir,'testdata', 'testdata', folder_name)
+        print test_data_dir
+
+        # ([failure_messages], {measurement_name: measurement_value})
+        failures = []
+        measurements = {}
+
+        _failures, _measurements = analyze_battery_stats(test_data_dir)
+        failures.extend(_failures)
+        measurements.update(_measurements)
+
+        _failures, _measurements = analyze_graphic_stats(test_data_dir)
+        failures.extend(_failures)
+        measurements.update(_measurements)
+
+        _failures, _measurements = analyze_execution_time(test_data_dir)
+        failures.extend(_failures)
+        measurements.update(_measurements)
+
+        ElementTree.SubElement(element, 'system-out').text = "\n".join(['<measurement><name>%s</name><value>%s</value></measurement>' % (k, v) for k, v in measurements.iteritems()])
+        ElementTree.SubElement(element, 'failure').text = '\n'.join(failures)
```

Comment:
Take into account that `failures` may be empty. In that case, the `failure` XML element (`<failure/>`) shouldn't be generated or the test case will be considered failed.